isis/tests/BundleSettingsTests.cpp

```diff
@@ -0,0 +1,825 @@
+#include "BundleSettings.h"
+
+#include <algorithm>
+
+#include <QBuffer>
+#include <QByteArray>
+#include <QString>
+#include <QList>
+#include <QPair>
+#include <QDomDocument>
+#include <QXmlStreamWriter>
+
+#include "BundleObservationSolveSettings.h"
+#include "BundleTargetBody.h"
+#include "IException.h"
+#include "MaximumLikelihoodWFunctions.h"
+
+#include "gmock/gmock.h"
+
+bool observationSettingsComparison(
+      BundleObservationSolveSettings m,
+      BundleObservationSolveSettings n
+) {
+  return (m.instrumentId() == n.instrumentId());
+}
+
+::testing::AssertionResult assertObservationSettingsEqual(
+      const char* m_expr,
+      const char* n_expr,
+      BundleObservationSolveSettings m,
+      BundleObservationSolveSettings n
+) {
+  if (observationSettingsComparison(m, n)) return ::testing::AssertionSuccess();
+
+  return ::testing::AssertionFailure() << m_expr << " and " << n_expr
+        << " are different because they have different InstrumentIds ("
+        << m.instrumentId().toStdString() << " and "
+        << n.instrumentId().toStdString() << ")";
+}
+
+QDomDocument saveToQDomDocument(BundleSettings &settings) {
```

Comment:
Can this go away after all QString replacements...?

Reply:
I decided to just make this use the QString and then use it everywhere because it still takes about 4 lines of code to convert to a QDomDocument